src/lib/Syntax.hs

```diff
@@ -1510,6 +1529,30 @@ pattern NothingAtom ty = DataCon MaybeDataDef [ty] 0 []
 pattern JustAtom :: Type -> Atom -> Atom
 pattern JustAtom ty x = DataCon MaybeDataDef [ty] 1 [x]
 
+pattern NestOne :: a -> Nest a
+pattern NestOne x = Nest x Empty
+
+pattern BinderAnn :: a -> BinderP a
+pattern BinderAnn x <- ((\case Ignore   ann  -> ann
+                               Bind (_:>ann) -> ann) -> x)
+  where BinderAnn x = Ignore x
```

Comment:
Can we remove this direction (i.e. make it a pattern, but not an expression)? I would argue that `BinderAnn x` is strictly less readable than `Ignore x` (and even more verbose!)

Reply:
Yes, good idea